internal/lint/subscribe.go

```diff
@@ -85,24 +80,37 @@ func Subscriber(input *SubscriberInput) {
 		Ω.Expect(errors.Is(actualErr, expectedErr)).To(Ω.BeTrue())
 
 		// only one (first received) message handled - before error is returned:
-		Ω.Expect(extractData(handler.Received)).To(Ω.HaveLen(1))
+		Ω.Expect(handler.Len()).To(Ω.Equal(1))
 	})
 }
 
 type mockHandler struct {
-	Err      error // error to return after handling (first) message
-	Received []bps.SubMessage
+	mu   sync.RWMutex
+	data []string
+
+	Err error // error to return after handling (first) message
 }
 
 func (h *mockHandler) Handle(msg bps.SubMessage) error {
-	h.Received = append(h.Received, msg)
+	h.mu.Lock()
+	h.data = append(h.data, string(msg.Data()))
+	h.mu.Unlock()
+
 	return h.Err
 }
 
-func extractData(msgs []bps.SubMessage) [][]byte {
-	data := make([][]byte, 0, len(msgs))
-	for _, msg := range msgs {
-		data = append(data, msg.Data())
-	}
+func (h *mockHandler) Data() []string {
+	h.mu.RLock()
+	defer h.mu.RUnlock()
+
+	data := make([]string, len(h.data))
+	_ = copy(data, h.data)
```

Comment:
ultra-minor, but you don't even need to copy this. `appends` to data (if mutative) will assign a new memory address, simply `return h.data` is fine (as long as protected by Rlock at the time of calling)

Reply:
Well, not always? Go's append allocates 2x cap, IIRC, and then increments size (used/len) so it's not always changing slice ptr?

If being 100% correct, that could probably be done as `return h.data[:]`?

Reply:
if cap is large enough it appends to data but it doesn't change it memory address, the pointer stays the same, hence no race